packages/react/src/components/DetailsList/DetailsList.base.tsx

```diff
@@ -597,8 +597,10 @@ const DetailsListInner: React.ComponentType<IDetailsListInnerProps> = (
     onBlur: focusZoneProps && focusZoneProps.onBlur ? focusZoneProps.onBlur : onBlur,
   };
 
+  const FinalGroupedList = groups && groupProps?.groupedListAs ? groupProps.groupedListAs : GroupedList;
```

Comment:
```ts
const FinalGroupedList = groups && groupProps?.groupedListAs ? composeComponentAs(groups && groupProps.groupedListAs, GroupedList) : GroupedList;
```

Reply:
What about?:

```ts
const FinalGroupedList = groups && groupProps?.groupedListAs ? composeComponentAs(groupProps.groupedListAs, GroupedList) : GroupedList;
```

shouldn't need to test the existence of `groups` inside `composeComponent` unless I'm missing something?